libtransmission/torrent.cc

```diff
@@ -2453,6 +2453,37 @@ bool renameArgsAreValid(std::string_view oldpath, std::string_view newname)
         !tr_strvContains(newname, TR_PATH_DELIMITER);
 }
 
+bool renameNoCollision(tr_torrent const* tor, std::string_view oldpath, std::string_view newname)
```

Comment:
I think I would fold this into `renameArgsAreValid()` instead of making it a separate function.